grails-app/domain/org/pih/warehouse/order/OrderItem.groovy

```diff
@@ -101,16 +106,40 @@ class OrderItem implements Serializable, Comparable<OrderItem> {
         actualDeliveryDate(nullable: true)
     }
 
+    String getUnitOfMeasure() {
+        if (quantityUom) {
+            return "${quantityUom?.code}/${quantityPerUom as Integer}"
+        }
+        else {
+            def g = ApplicationHolder.application.mainContext.getBean( 'org.codehaus.groovy.grails.plugins.web.taglib.ApplicationTagLib' )
+            return "${g.message(code:'default.ea.label').toUpperCase()}/1"
+        }
+    }
+
+    Integer getQuantityPerUom() {
+        return this.quantityPerUom?:1
+    }
+
     def getShippedShipmentItems() {
         return shipmentItems.findAll { it.shipment.currentStatus >= ShipmentStatusCode.SHIPPED }
     }
 
+    def getQuantityShipped() {
+        def quantityShipped = shippedShipmentItems?.sum { it?.quantity }?:0
+        return quantityShipped / (quantityPerUom?:1)
+    }
+
+    def getQuantityReceived() {
+        def quantityReceived = shippedShipmentItems?.sum { it?.quantityReceived() }?:0
+        return quantityReceived / (quantityPerUom?:1)
+    }
+
     String getOrderItemType() {
-        return (product) ? "Product" : (category) ? "Category" : "Unclassified"
+        return "Product"
     }
 
     Integer quantityFulfilled() {
```

Comment:
I guess this is redundant now.

Reply:
I plan to remove it at some point, I just don't feel like refactoring now.